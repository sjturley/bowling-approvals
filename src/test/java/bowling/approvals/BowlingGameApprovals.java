/*
 * This Java source file was generated by the Gradle 'init' task.
 */
package bowling.approvals;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class BowlingGameApprovals {
    @Test
    void approvingAllBowlingGameRules() {
        BowlingGame bowlingGame = new BowlingGame();

        // Frame 1 (0)
        bowlingGame.roll(0);
        bowlingGame.roll(0);
        // Frame 2 (8)
        bowlingGame.roll(2);
        bowlingGame.roll(6);
        // Frame 3 (6) - 14
        bowlingGame.roll(5);
        bowlingGame.roll(1);
        // Frame 4 (11) - 25
        bowlingGame.roll(5);
        bowlingGame.roll(5);
        // Frame 5 (1) - 26
        bowlingGame.roll(1);
        bowlingGame.roll(0);
        // Frame 6 (16) - 42
        bowlingGame.roll(10);
        // Frame 7 (6) - 48
        bowlingGame.roll(5);
        bowlingGame.roll(1);
        // Frame 8 (0) - 48
        bowlingGame.roll(0);
        bowlingGame.roll(0);
        // Frame 9 (0) - 48
        bowlingGame.roll(0);
        bowlingGame.roll(0);
        // Frame 10 (30) - 78
        bowlingGame.roll(10);
        bowlingGame.roll(10);
        bowlingGame.roll(10);

        Approvals.verify(bowlingGame.toJson());
    }
}
