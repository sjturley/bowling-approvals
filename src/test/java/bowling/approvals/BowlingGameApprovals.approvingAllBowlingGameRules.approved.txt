{
    frames: [
        {"firstRoll": 0, "secondRoll": 0, frameScore: 0, runningScore: 0}
    ]
}